m02_setup/01_first_steps.rst

```diff
@@ -0,0 +1,190 @@
+  and examining their error messages. To be precise, there is the terminal
+  emulator window and inside it another program, `shell`, which reads
+  user commands and acts according to them. If you are new to this, read the
+  `"Learning the shell" LinuxCommand.org <http://linuxcommand.org/lc3_learning_the_shell.php>`_
+  tutorial parts 1-6.
+
+Software prerequisites
+----------------------
+
+The first step to start authoring a course is to install some required software on your computer. The first two software you need are `git <https://git-scm.com/>`_ and `make <https://www.gnu.org/software/make/>`_. Sometimes, these two software are installed by default, but if the **git** and **make** are not installed by default, you can install them by running the :ref:`commands below <install-git>` in your machine. Otherwise, you can consult the official documentation on how to `install git <https://git-scm.com/book/en/v2/Getting-Started-Installing-Git>`_, and how to `install make <https://pkgs.org/download/build-essential>`_ in your OS. 
+
+.. warning:: 
+  if you are using an Aalto's computer, you will need administrator rights to install the required software; otherwise, you can ask an administrator to install them for you. `Aalto IT services <https://www.aalto.fi/en/service-entities/it-services>`_ have administrator rights for all the Aalto's computers. 
+
+.. code-block:: bash
+  :name: install-git
+  :caption: 1- Install Git
+
+  sudo apt update
+  sudo apt install git
+
+.. code-block:: bash
+  :caption: 2- Verify git version
+
+  git --version
+
+.. code-block:: bash
+  :caption: 3- Install build-essential, which includes make.
+
+  sudo apt update
+  sudo apt install build-essential
+
+.. code-block:: bash
+  :caption: 4- Verify make version
+
+  make --version
+
+::::
+
+All other software runs inside `Docker <https://www.docker.com/>`_ containers.
+For short, Docker is the new software platform for running A+ and software
+related to it. This course will discuss Docker later.
+
+Install Docker for Mac/Windows or docker-ce & docker-compose for Linux.
+
+- `Docker Community edition <https://docs.docker.com/engine/installation/>`_
+- `Docker Compose <https://docs.docker.com/compose/install/>`_
+
+You also need to install a Text editor (`VS Code <https://code.visualstudio.com/>`_  is recommended). You can find more information about the most convenient text editor to create *RST* documents :ref:`at the end of this module <text-editor-and-ides>`.
+
+Cloning this codebase with git
+------------------------------
+
+A good way to start is to clone the `Aplus course Template <https://github.com/apluslms/aplus-course-template>`_ with git and begin to work on it. In Aalto, you could use `version.aalto.fi repo <https://version.aalto.fi/gitlab/course/aplus-manual>`_. Please note that if you don't have access, then you can request access to the *Aalto IT services*. In that case, you probably should be part of :code:`apluslms-cs@aalto.fi` email list too.
```

Comment:
> Please note that if you don't have access, then you can request access to the *Aalto IT services*

Wrong. Access should be requested from the EDIT team, but since the aplus-manual repo is public in github.com, teachers don't really need access to the private version.aalto.fi repo.

Reply:
Access to the (Aalto-internal) `apluslms-cs` email list is also requested from the EDIT team.